### ISSUE TITLE
How to put a context inside custom control onClick function argument?

### ISSUE THREAD

**Issue by Gonzalosanto** [geoman-io/leaflet-geoman#980]
Hi! I have an issue about creating a custom control to upload layers data to an endpoint at NodeJS API REST server.

I don't have any idea why this onClick handler gets a global PointerEvent object. Do you have any idea how to pass an specific layer data (marker or polygon when you finished to created it) as argument inside the function upload?

An approach I found was searching inside global object the data I was looking for, but I don't like this solution.

 [DEMO SCRIPT](https://jsfiddle.net/gnrk3bju/2/)
 
 I'm working with plugin in vanilla JS using CDN Script and Stylesheet.

**Comment by Falke-Design** [comment]
You can get all drawn layers with `map.pm.getGeomanDrawLayers()` or you add the created layers in the `pm:create` event to an array.
When the upload control is clicked you can use the array and send them to the API.

You need to store the data in a global variable because from where should the onClick function know which data it should take?

**Comment by Gonzalosanto** [comment]
> You can get all drawn layers with `map.pm.getGeomanDrawLayers()` or you add the created layers in the `pm:create` event to an array.
> When the upload control is clicked you can use the array and send them to the API.
> 
> You need to store the data in a global variable because from where should the onClick function know which data it should take?

Yes, what I'm doing in my script is loading with pm:create all the layers I draw and it is stored inside a global object inside global window object. But, the question is how to get this object without getting a PointerEvent object when onClick is triggered... Because when I get this PointerEvent object I have to expand it and search my array of created layers.

**Comment by Falke-Design** [comment]
Sorry but I don't understand the problem. Simply use the global variable instead of the payload of the function:

```
var yourDataLayers = [];

var customOpts = {
          name: 'Upload Button',
          block: 'custom',
          title: 'Subir Referencias',
          toggle:false,
          onClick: function upload(IGNORE_THIS){
            console.log(yourDataLayers);       
            fetch('/url/of/endpoint', {
            	method: 'POST',
              body: yourDataLayers 
            })
          },
          icon: "item"
        }
```
